keras_nlp/tokenizers/word_piece_tokenizer.py

```diff
@@ -256,6 +256,15 @@ def get_config(self) -> Dict[str, Any]:
         return config
 
     def tokenize(self, inputs):
+        # Check if Input is Scalar or Not
+        if not isinstance(inputs, tf.Tensor) or not isinstance(
```

Comment:
I think we could just convert to tensor generally here, rather then converting only to check the rank. Add something like this first...

```
        if not isinstance(inputs, (tf.Tensor, tf.RaggedTensor)):
            inputs = tf.convert_to_tensor(inputs)
```

Then you can just run `scalar_input = tf.convert_to_tensor(inputs).shape.rank == 0` assuming tensor.

Reply:
Got it this is indeed a better way, thanks!